### ISSUE TITLE
Deploy dev/test cluster?

### ISSUE THREAD

**Issue by reisub0** [redpanda-data/helm-charts#229]
Hello, I'm new to Redpanda and I'm trying to figure out how to deploy a test cluster to see if my application works with it as a drop-in-replacement.

I'm trying to deploy Redpanda on extremely resource-constrained VMs. I was initially surprised that it simply would not deploy (because the default limits were too high).

Investigating further, I found that the `values.yaml` file has the following comment:
```
  # The default values below are what should work for a development environment.
  # Production-level values and other considerations are provided in comments
  # if those values are different from the default.
```
and:
```
    container:
      # Minimum memory count for each Redpanda broker
      # If omitted, the min value will equal the max value (requested resources defaults to limits)
      # Equivalent to: resources.requests.memory
      # For production: 10Gi or greater
      # min: 2.5Gi

```

This makes sense. 

However, as of now, there's no way to change these values to dev/test values like a 0.5Gi test cluster, as the helm validation fails it with the following error:

```
Error: INSTALLATION FAILED: execution error at (redpanda/templates/_helpers.tpl:341:84):
81 is below the minimum recommended value for Redpanda
```

It definitely makes sense to have strong defaults and enforce best practices, but as far as my understanding goes, it is very much possible to run a Redpanda node with just 0.5 Gi of memory allocated. It doesn't make sense to take away power from the user and make it completely impossible to do so from the Helm deployment.

I also found some discussion about this at https://github.com/redpanda-data/redpanda/issues/175 but that's just for setting the `developer_mode` config parameter, which would probably be required in this case as well, but setting this to true does not help me bypass the check, given that a comment says:
```
# This section allows modifying various Redpanda settings not covered in other sections above.
# These values do not pertain to the kubernetes objects created with helm.
# Instead these parameters get passed directly to the Redpanda binary at startup.
# See https://docs.redpanda.com/docs/cluster-administration/configuration/
```

**Comment by carlreid** [comment]
I totally agree. I wanted to deploy in a test environment with a lower amount of memory too, but ended up just accepting the lower limit of 2.5Gi for now.

In general, I think all of the values in the example feel like they're plucked out of the air with no context. Like how many topics/events per second would 2.5Gi and 1 CPU support? In our case the instance is only for my team's events which is very low right now, and as such, that many resources seems a bit crazy.

I think it would be great if these values could be calculated to some extent based on some formula.

**Comment by joejulian** [comment]
You're totally right and we're working on a change for addressing that. Right now it seems to be a mix of "you must do this for production" and "this is a development installation" which is not at all what we want it to be.

The upcoming change will make all the defaults "production" ready, for some specific (and documented) definition of production. A couple examples files will have suggestions for minimal "api exercising" and "mild development" configurations. We should see the PR for this opening very soon.

In the mean time, I'll double check with the core team to find out if there is an actual minimum and if it'll run with less than the 2GiB per core that's documented I'll open a PR to change that check.

**Comment by joejulian** [comment]
As an aside, @carlreid , have you seen https://redpanda.com/blog/sizing-redpanda-cluster-best-practices ? I think it might answer some of what you're asking.

**Comment by carlreid** [comment]
I hadn't actually @joejulian, it certainly helps to get a better idea of the resources needed 👍 In my case I can definitely reduce the resources by quite a lot 😁 Though then I'd still be blocked by the Helm validation mentioned here.

The article also doesn't really state the problems of bring CPU/mmory values really low. It mentions about being bottlenecked but how? Are we talking a couple milliseconds, or seconds to retrieve/publish events? For my current use case it's alright if it isn't blazing fast, which of course is one of Redpanda's attractive points, and that performance is nice to have available for when I need it.

**Comment by reisub0** [comment]
I personally consider this a bug in the chart. Helm could (and should!) warn the user that this amount of resources isn't recommended for a production cluster and it's not supported/might cause underwhelming performance etc, but at the same time, it **shouldn't completely forbid** the user from making any cluster that would potentially work. 

If hoops need to be jumped through to enable such a dev/test workflow in resource constrained environments, it should be documented, but the steps shouldn't be "patch" the helm chart to remove the validation entirely, which is what I had to do.
